js_modules/dagit/packages/core/src/app/AppProvider.tsx

```diff
@@ -205,6 +206,7 @@ export const AppProvider: React.FC<AppProviderProps> = (props) => {
                       </CustomConfirmationProvider>
                       <CustomTooltipProvider />
                       <CustomAlertProvider />
+                      <AssetRunLogObserver />
```

Comment:
I see, it renders `SingleRunLogObserver` which subscribes using our hooks. It's a shame because this ideally wouldn't need to live in the React tree at all.